### PR TITLE
Feature/table component

### PR DESCRIPTION
**Pull request by clementprdhomme**
This PR adds a table component, fully accessible, with sortable columns and an optional search field.

Some notes:
- [x] Even if the PR contains the styles of the search field, it won't appear as on the design because it needs to be placed within another component ([Pivotal task](https://www.pivotaltracker.com/story/show/130032081)).
- [x] Currently, the limit of results per page is not implemented and needs design as Dani expressed his wish to have a "show more" button instead ([Pivotal task](https://www.pivotaltracker.com/story/show/130032281)).
- [x] The API needs to be updated before the component can be widely used on the app. The list of specifications has been written down on a [Pivotal task](https://www.pivotaltracker.com/story/show/130031033).
- [x] The links to view, edit and delete an element will be styled in another component but also need the API to return the correct format ([Pivotal task](https://www.pivotaltracker.com/story/show/130032623)).

### CODE REVIEW

**Comment by tiagojsag** [comment]
@clementprdhomme is this still relevant? maybe you should assign it to someone else instead

**Comment by clementprdhomme** [comment]
I just resolved the conflict with `develop`. I'll assign @andresgnlez to this PR so he can review.

**Comment by andresgnlez** [comment]
image

@clementprdhomme app is failing on build because you are trying to use `$color-6` in a shared component and this CSS variable is not defined in your current settings file (`application-theme-1.scss`). Probably there will be the same error in `application-theme-2.scss`

Could you please review this?

File: app/assets/stylesheets/components/shared/c-tables.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          width: 0;
+          height: 0;
+          margin-left: 12px;
+          transform: translateY(-3px) rotate(-135deg);
+          border-top: 4px solid currentColor;
+          border-right: 4px solid transparent;
+          border-bottom: 4px solid transparent;
+          border-left: 4px solid currentColor;
+          content: '';
+        }
```
Comment (andresgnlez) [inline app/assets/stylesheets/components/shared/c-tables.scss]: Perhaps you could use a triangle helper here, I think it's worth it.

**Comment by andresgnlez** [comment]
Just a few comments. Well done!

**Comment by clementprdhomme** [comment]
I think we can merge this PR without having the "show more" button ready for now.

@tiagojsag Can you tell me how I could build the links to the show/edit/delete pages from the front? Once the links are real, this PR is ready to merge!

**Comment by clementprdhomme** [comment]
The PR needs to be rebased. Some of the files like the CSS ones have been merged to `develop` and a bit modified.

**Comment by tiagojsag** [comment]
@clementprdhomme rebase to merge or close, please

**Pull request merged**